`time.sleep(1000).wait()` from `main` cause compilation file

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, "Garuda Linux"
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 3800X 8-Core Processor
CC version: cc (GCC) 12.1.0

getwd: /home/yuart/Projects/DiscordLinksCollector
vmodules: /home/yuart/.vmodules
vroot: /home/yuart/Projects/v
vexe: /home/yuart/Projects/v/v
vexe mtime: 2022-08-02 11:56:15
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 b0c32e0.857e047

Git version: git version 2.37.1
Git vroot status: weekly.2022.31-6-g857e047d (1 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v run .`
```v
module main

import net.http
import zztkm.vdotenv
import os
import json
import time

fn main() {
	vdotenv.load()

	mut links_request := http.Request{
		method: .get
		url: 'https://discord.com/api/v10/guilds/${os.getenv('GUILD_ID')}/messages/search?channel_id=${os.getenv('CHANNEL_ID')}&has=link'
	}

	links_request.add_header(.authorization, os.getenv('AUTHORIZATION'))

	links_response := links_request.do()?
	time.sleep(1000).wait()
	links_data := json.decode(LinksData, links_response.body)?

	mut links := []string{}

	for message in links_data.messages {
		links << message[0].content
	}

	links_count := links_data.total_results
	mut current_offset := 0

	for current_offset + 25 < links_count {
		current_offset += 25

		mut links_request_1 := http.Request{
			method: .get
			url: 'https://discord.com/api/v10/guilds/${os.getenv('GUILD_ID')}/messages/search?channel_id=${os.getenv('CHANNEL_ID')}&has=link&offset=$current_offset'
		}

		links_request_1.add_header(.authorization, os.getenv('AUTHORIZATION'))

		links_response_1 := links_request_1.do()?
		time.sleep(1000).wait()
		links_data_1 := json.decode(LinksData, links_response_1.body)?

		for message in links_data_1.messages {
			links << message[0].content
		}
	}

	println(links.len)
	println(current_offset)
}

struct LinksData {
	total_results u32
	messages      [][]Message
}

struct Message {
	content string
}

```

**What did you expect to see?**

Compiled program or error message from V


**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 000004f5
tcc: error: file 'crt1.o' not found
/tmp/v_1000/DiscordLinksCollector.8055781203055421562.tmp.c:29618: warning: implicit declaration of function 'void_wait'
tcc: error: undefined symbol 'void_wait'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1114,7 +1114,7 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) 
 pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 	left_type := c.expr(node.left)
 	if left_type == ast.void_type {
-		c.error("cannot call a method using an invalid expression", node.pos)
+		c.error('cannot call a method using an invalid expression', node.pos)
 		return ast.void_type
 	}
 	c.expected_type = left_type

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1113,6 +1113,10 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) 
 
 pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 	left_type := c.expr(node.left)
+	if left_type == ast.void_type {
+		c.error("cannot call method using an invalid expression", node.pos)
+		return ast.void_type
+	}
 	c.expected_type = left_type
 	mut is_generic := left_type.has_flag(.generic)
 	node.left_type = left_type

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1114,7 +1114,7 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) 
 pub fn (mut c Checker) method_call(mut node ast.CallExpr) ast.Type {
 	left_type := c.expr(node.left)
 	if left_type == ast.void_type {
-		c.error("cannot call method using an invalid expression", node.pos)
+		c.error("cannot call a method using an invalid expression", node.pos)
 		return ast.void_type
 	}
 	c.expected_type = left_type

--- a/vlib/v/checker/tests/void_method_call.out
+++ b/vlib/v/checker/tests/void_method_call.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/void_method_call.vv:5:17: error: cannot call a method using an invalid expression
+    3 |
+    4 | fn main() {
+    5 |     simple_fn().method()
+      |                 ~~~~~~~~
+    6 | }

--- a/vlib/v/checker/tests/void_method_call.vv
+++ b/vlib/v/checker/tests/void_method_call.vv
@@ -1,0 +1,6 @@
+fn simple_fn() {
+}
+
+fn main() {
+    simple_fn().method()
+}

